File: backend-spring/src/main/java/com/sourcelens/module/execution/entity/ExecutionTask.java
package com.sourcelens.module.execution.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("execution_tasks")
public class ExecutionTask {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private Long repositoryId;

    private String taskType;

    private String sourceType;

    private Long sourceId;

    private String status;

    private String currentStep;

    private Long currentAttemptId;

    private Integer progress;

    private String errorMessage;

    private Long createdBy;

    private LocalDateTime startedAt;

    private LocalDateTime finishedAt;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: web-console/src/api/executionTask.ts
import client from './client'
import type { Result } from './client'

export interface ExecutionTask {
  id: number
  projectId: number
  repositoryId: number | null
  taskType: string
  sourceType: string | null
  sourceId: number | null
  status: string
  currentStep: string | null
  currentAttemptId: number | null
  progress: number
  errorMessage: string | null
  createdBy: number
  startedAt: string | null
  finishedAt: string | null
  createdAt: string
  updatedAt: string
}

export interface ExecutionStep {
  id: number
  taskId: number
  attemptId: number | null
  stepKey: string
  stepName: string
  status: string
  logSummary: string | null
  errorMessage: string | null
  startedAt: string | null
  finishedAt: string | null
  createdAt: string
  updatedAt: string
}

export interface ExecutionAttempt {
  id: number
  taskId: number
  attemptNo: number
  status: string
  currentStep: string | null
  errorMessage: string | null
  startedAt: string | null
  finishedAt: string | null
  createdAt: string
  updatedAt: string
}

export interface ExecutionLog {
  id: number
  taskId: number
  attemptId: number | null
  stepKey: string | null
  level: string
  message: string
  createdAt: string
}

export interface ExecutionTaskDetail {
  task: ExecutionTask
  attempts: ExecutionAttempt[]
  steps: ExecutionStep[]
  logs: ExecutionLog[]
}

export interface PageResult<T> {
  items: T[]
  page: number
  pageSize: number
  total: number
}

export const executionTaskApi = {
  list: (projectId: number, page = 1, pageSize = 20) =>
    client.get<Result<PageResult<ExecutionTask>>>(`/projects/${projectId}/execution-tasks`, {
      params: { page, pageSize },
    }),

  detail: (projectId: number, taskId: number) =>
    client.get<Result<ExecutionTaskDetail>>(`/projects/${projectId}/execution-tasks/${taskId}`),

  detailBySource: (projectId: number, sourceType: string, sourceId: number) =>
    client.get<Result<ExecutionTaskDetail>>(`/projects/${projectId}/execution-tasks/source/${sourceType}/${sourceId}`),

  cancel: (projectId: number, taskId: number) =>
    client.post<Result<ExecutionTaskDetail>>(`/projects/${projectId}/execution-tasks/${taskId}/cancel`),
}
